core/src/main/scala/org/http4s/StaticFile.scala

```diff
@@ -106,7 +103,11 @@ object StaticFile {
                 Some(
                   Response(
                     headers = headers,
-                    body = readInputStream[F](F.pure(inputStream), DefaultBufferSize)
+                    entity = {
+                      val body =readInputStream[F](F.pure(inputStream), DefaultBufferSize)
+                      if (len >= 0) Entity.TrustMe(body, len)
```

Comment:
Something that has always bothered me: what if the file gets truncated while we're reading it?

Reply:
Betrayed trust.